src/hibernate-resume/hibernate-resume.c

```diff
@@ -109,31 +118,44 @@ static int run(int argc, char *argv[]) {
 
         umask(0022);
 
-        if (!in_initrd())
+        if (!in_initrd() && !arg_clear_efi)
+                return log_error_errno(SYNTHETIC_ERRNO(ENOTRECOVERABLE),
+                                       "Not running in initrd, refusing to initiate resume from hibernation.");
+        if (in_initrd() && arg_clear_efi) {
+                log_info("Running in initrd, --clear-efi has no effect, exiting.");
                 return 0;
+        }
 
-        if (argc > optind) {
-                arg_info.device = argv[optind];
+        if (argc <= optind || arg_clear_efi) {
```

Comment:
If `--clear-efi` is specified, extra arguments will be ignored. Please refuse when an extra argument is specified, or at least warn about that.